requests/models.py

```diff
@@ -293,15 +293,18 @@ def __init__(self):
         self.hooks = default_hooks()
         #: integer denoting starting position of a readable file-like body.
         self._body_position = None
+        #: boolean denoting whether or not to discard cookies from the response.
+        self.discard_cookies = False
```

Comment:
As discussed, I'd rather not have this flag. It'd be better to be able to pass it directly into the adapter than to force the request to carry it around.